smother/control.py

```diff
@@ -41,7 +46,7 @@ def start(self):
 
     def save_context(self, label):
         self.data[label] = {
-            key: sorted(map(int, val.keys()))
+            self._normalize_path(key): sorted(map(int, val.keys()))
```

Comment:
Ok this thread is my only remaining concern. I'd prefer that we find a way to do away with the `--reative-paths` flag since it complicates the user-facing interface and seems too similar to what the `paths` config section already does. Let's refine the terminology and goals here. There are 3 phases of the .smother file lifecycle: *collection* time, *combine* time, and *report* time. I think they should work like this

* At *collect* time, we always do what coverage does (we dump absolute paths to a file).
* At *combine time*, we've already gathered several .smother files onto a new machine. We can assume that this machine will be used during report time, and the files referenced in the smother report are available somewhere (we're often but not always in the root directory of the files in a .smother file). The user is required to configure their coveragerc file so that the path section normalizes files into correct paths on the current machine. If we use PathAliases the combined report will [very likely include](https://bitbucket.org/ned/coveragepy/src/7c3dddfe83893c51035c0be4f35b58d0379858ac/coverage/files.py?at=default&fileviewer=file-view-default#files.py-352) absolute paths, but that's ok. Alternatively we could implement our own PathAliases equivalent that doesn't convert to canonical paths, and instead treats the aliases in the coveragerc file more literally. 
* At *report time* we don't care whether the input data are absolute or relative paths, we just require that `os.path.exists(path)`. We don't apply any normalization, so we don't pay any speed penalties. 

This flow allows us to eliminate the `--relative-paths` flag, and the only thing we lose is the ability to force smother files that have relative paths. Relative paths are kind of nice if you're manually looking at the file, but not required for portability since you can always achieve the same thing with a `.coveragerc` and `smother combine`. I don't think that's a strong enough reason to add a new flag to smother.